src/services/createRouterFind.ts

```diff
@@ -24,11 +26,11 @@ export type RouterFind<
 
 export function createRouterFind<const TRoutes extends Routes>(routes: TRoutes): RouterFind<TRoutes> {
 
-  return <TRoutes extends Routes, TSource extends Url | RoutesKey<TRoutes>>(
+  return <TRoutes extends Routes, TSource extends RoutesKey<TRoutes>>(
     source: TSource,
     params: Record<PropertyKey, unknown> = {},
   ): ResolvedRoute | undefined => {
-    const resolve = createRouterResolve(routes)
+    const resolve = createRouterResolve(routes as Route<string, Path, Query, false>[])
```

Comment:
this one was annoying. I had to widen routes to make `resolve` happy. It kept complaining that source wasn't satisfies `Url | RoutesKey<T>`